cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -0,0 +1,233 @@
+
+    private MManagerHolder() {
+      // allowed to do nothing
+    }
+
+    private static final CMManager INSTANCE = new CMManager();
+  }
+
+  /**
+   * we should not use this function in other place, but only in IoTDB class
+   * @return
+   */
+  public static MManager getInstance() {
+    return CMManager.MManagerHolder.INSTANCE;
+  }
+
+  @Override
+  public String deleteTimeseries(String prefixPath) throws MetadataException {
+    cacheLock.writeLock().lock();
+    mRemoteMetaCache.removeItem(prefixPath);
+    cacheLock.writeLock().unlock();
+    return super.deleteTimeseries(prefixPath);
+  }
+
+  @Override
+  public void deleteStorageGroups(List<String> storageGroups) throws MetadataException {
+    cacheLock.writeLock().lock();
+    for (String storageGroup : storageGroups) {
+      mRemoteMetaCache.removeItem(storageGroup);
+    }
+    cacheLock.writeLock().unlock();
+    super.deleteStorageGroups(storageGroups);
+  }
+
+  @Override
+  public TSDataType getSeriesType(String path) throws MetadataException {
+    try {
+      cacheLock.readLock().lock();
+      MeasurementMeta measurementMeta = mRemoteMetaCache.get(path);
+      return measurementMeta.getMeasurementSchema().getType();
+    } catch (IOException e) {
+      //do nothing
+    } finally {
+      cacheLock.readLock().unlock();
+    }
+    return super.getSeriesType(path);
+  }
+
+  @Override
+  public MeasurementSchema[] getSchemas(String deviceId, String[] measurements) throws MetadataException {
+    try {
+      return super.getSchemas(deviceId, measurements);
+    } catch (MetadataException e) {
+      // some measurements not exist in local
+      // try cache
+      MeasurementSchema[] measurementSchemas = new MeasurementSchema[measurements.length];
+      boolean allSeriesExists = true;
+      cacheLock.readLock().lock();
+      for (int i = 0; i < measurements.length; i++) {
+        try {
+          MeasurementMeta measurementMeta = mRemoteMetaCache.get(deviceId + measurements[i]);
+          measurementSchemas[i] = measurementMeta.getMeasurementSchema();
+        } catch (IOException ex) {
+          // not all cached, pull from remote
+          allSeriesExists = false;
+          break;
+        }
+      }
+      cacheLock.readLock().unlock();
+      if (allSeriesExists) {
+        return measurementSchemas;
+      }
+
+      pullSeriesSchemas(deviceId, measurements);
+
+      // try again
+      boolean allExist = true;
+      cacheLock.readLock().lock();
+      for (int i = 0; i < measurements.length; i++) {
+        try {
+          MeasurementMeta measurementMeta = mRemoteMetaCache.get(deviceId + measurements[i]);
+          measurementSchemas[i] = measurementMeta.getMeasurementSchema();
+        } catch (IOException ex) {
+          allExist = false;
+          break;
+        }
+      }
+      cacheLock.readLock().unlock();
+      if (!allExist) {
+        throw new MetadataException(deviceId + " has some mesurements not found");
```

Comment:
It would be better to show which series is not found and just throw PathNotExistException.

Reply:
ok